src/fsharp/NameResolution.fs

```diff
@@ -4445,21 +4445,12 @@ let rec GetCompletionForItem (ncenv: NameResolver) (nenv: NameResolutionEnv) m a
                     else Seq.empty)
             
             // Look for values called 'id' that accept the dot-notation 
-            let values, isItemVal = 
-                (if nenv.eUnqualifiedItems.ContainsKey(id) then 
-                         // v.lookup : member of a value
-                  let v = nenv.eUnqualifiedItems.[id]
-                  match v with 
-                  | Item.Value x -> 
-                      let typ = x.Type
-                      let typ = if x.BaseOrThisInfo = CtorThisVal && isRefCellTy g typ then destRefCellTy g typ else typ
-                      (ResolvePartialLongIdentInTypeForItem ncenv nenv m ad false rest item typ), true
-                  | _ -> Seq.empty, false
-                 else Seq.empty, false)
-            
-            yield! values
-
-            if not isItemVal then 
+            match Map.tryFind id nenv.eUnqualifiedItems with
+            | Some (Item.Value x) ->
```

Comment:
This allocates, same as the other PR - are we sure this isn't a problem?

Reply:
We're not sure.  

That said, ephemeral allocations that don't survive Gen0 are very, very cheap